### DISCUSSION TITLE
The program crashes in the last epoch's validation in multi-gpu (DDP) case

### DISCUSSION THREAD

**Discussion by bardia-az** [ultralytics/yolov5#5557] [Q&A]
I am training yolov5x6 model on a Tesla V100-SXM2-32GB GPU. The used command is:

python -m torch.distributed.launch --nproc_per_node 4 train.py --epochs 2 --freeze 5 --weights yolov5x6.pt --data data/coco.yaml --name test --batch-size 64 --img 1024 --exist-ok --hyp data/hyps/hyp.scratch-p6.yaml --device 0,1,2,3 --adam

In the last epoch's validation, the program crashes and gives this error:

terminate called after throwing an instance of 'std::system_error'  
what():  Connection reset by peer

Do you have any idea what's the reason for that? Does it have something to do with the fact that GPU0 is still running but the other GPUs have finished their job?

**Comment by glenn-jocher** [comment]
@bardia-az 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. We've created a few short guidelines below to help users provide what we need in order to get started investigating a possible problem. In general though you should run all DDP commands in our Docker image for best results, and you should also use the newer `torch.distributed.run` command.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃
